src/xmipp/libraries/reconstruction_adapt_cuda/volume_halves_restoration_gpu.h

```diff
@@ -0,0 +1,159 @@
+#ifndef _PROG_VOLUME_HALVES_RESTORATION_GPU
+#define _PROG_VOLUME_HALVES_RESTORATION_GPU
+
+#include <chrono>
+#include <iostream>
+// #include <cuda_runtime_api.h>
+
+#include <core/xmipp_program.h>
+#include <core/xmipp_fftw.h>
+
+#include <data/mask.h>
+#include <data/numerical_tools.h>
+
+#include "reconstruction_cuda/cuda_cdf.h"
+#include "reconstruction_cuda/cuda_volume_restoration_kernels.h"
+#include "reconstruction_cuda/cuda_fft.h"
+#include "reconstruction_cuda/cuda_xmipp_utils.h"
+
+#include "reconstruction_cuda/cuda_volume_restoration_denoise.h"
+
+//comment this define to hide time measurements
+// #define TIME_MSR
```

Comment:
as mentioned earlier, please use time_utils and https://github.com/I2PC/xmippCore/blob/911e010f91ac785121d3b73d7cee2fe3930bae68/core/utils/time_utils.h#L160

Reply:
Removed time measurements.